### PR TITLE
fix: `as coins` map value serialization type now generates correct code

### LINKED ISSUE #985: Tact serializes map<Address, Int as coins> incorrectly
### Are you using the latest released (or pre-released, a.k.a. "next") version?

- [X] I'm using the latest Tact version

### Tact source code

```tact
import "@stdlib/deploy";

message ForwardPayload{
    to: map<Address, Int as coins>;
}
contract InvalidSerialization with Deployable {

    receive(m: ForwardPayload){
        foreach(addr, amount in m.to){
            dump(addr);
            dump(amount);
        }
    }
}
```

### Relevant Tact/build system log output

```shell
No errors thrown
```

### What happened?

When one tries to send "ForwardPayload" message, it ends with 9 error code, because the serialization and deserialization differ:
```typescript
export function storeForwardPayload(src: ForwardPayload) {
    return (builder: Builder) => {
        let b_0 = builder;
        b_0.storeUint(3264607134, 32);
        b_0.storeDict(src.to, Dictionary.Keys.Address(), Dictionary.Values.BigUint(124));
    };
}
```
and in contract code:
```func
    var ($addr, $amount, $fresh$flag_0) = __tact_dict_min_slice_int($m'to, 267, 257);
    while ($fresh$flag_0) {
        __tact_debug_address($addr, __gen_slice_string_fff0772d56d2a3f45220c12e605b1692cb65b9ef6f72a2e11c228fd2152f47e2(), "File contracts/test.tact:11:13:");
        __tact_debug_str(__tact_int_to_string($amount), __gen_slice_string_8ebac137868252bcaf2f59f69a6616ec14a1c79ceaf49369f36485c5fb48a8ec(), "File contracts/test.tact:12:13:");
        ($addr, $amount, $fresh$flag_0) = __tact_dict_next_slice_int($m'to, 267, $addr, 257);
```

### What did you expect?

I expected that the map values are serialized correctly both in the func code and TypeScript

### Steps to reproduce

```bash
git clone https://github.com/imartemy1524/tact-map-bug
cd tact-map-bug
npm i
npx blueprint build --all
npx blueprint test
```

### How do you run Tact?

Blueprint

### Anything else?

Source code: https://github.com/imartemy1524/tact-map-bug

### PR DESCRIPTION
**Pull request by Gusarich**
## Issue

Closes #985.

## Checklist

- [x] I have updated CHANGELOG.md
- [ ] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Comment by Gusarich** [comment]
will add tests in a moment

**Comment by anton-trunov** [comment]
What about the FunC part, shouldn't it use `load_coins` in this case?

**Comment by Gusarich** [comment]
> What about the FunC part, shouldn't it use `load_coins` in this case?

It does

**Comment by Gusarich** [comment]
> What about the FunC part, shouldn't it use `load_coins` in this case?

at first I thought that the FunC part works fine already, but just looked through the generated code for `as coins` maps and turned out that it's completely wrong and just uses 257-bit long integers. I will fix that too

**Comment by Gusarich** [comment]
but it'll be easier to fix after #941 gets merged

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `SendDefaultMode` send mode constant to the standard library: PR [#1010](https://github.com/tact-lang/tact/pull/1010)
 - Docs: initial semi-automated Chinese translation of the documentation: PR [#942](https://github.com/tact-lang/tact/pull/942)
 - The `replace` and `replaceGet` methods for the `Map` type: PR [#941](https://github.com/tact-lang/tact/pull/941)
+- `as coins` map value serialization type is now handled correctly: PR [#987](https://github.com/tact-lang/tact/pull/987)
```
Comment (anton-trunov) [inline CHANGELOG.md]: Let's move it to Fixed

File: src/generator/writers/writeStdlib.ts [hunk 2]
```diff
@@ -1161,7 +1161,8 @@ function genTactDictGet(
     value: ValType,
 ): void {
     const signatureKeyType = key === "uint" ? "int" : key;
-    const signatureRetType = value === "uint" ? "int" : value;
+    const signatureRetType =
```
Comment (anton-trunov) [inline src/generator/writers/writeStdlib.ts]: Looks like this should be refactored into a helper function (I should have done it in my previous refactoring)

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Awesome stuff! Just minor comments

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**